### PR TITLE
Feat  role filter

### PR DESCRIPTION
**Pull request by theogainey**
### Description
add role filter 
add toggleFilter onChange event to role filter buttons 
add slider hover state to filter buttons

### Spec

See Story: [191](https://sparkbox.atlassian.net/browse/FSA22V1-191)

### Validation

* [ ] This PR has code changes, and our linters still pass.
* [ ] This PR affects production code, so it was browser tested (see below).
* [ ] Changes are browser tested. This includes functionality, accessibility, responsiveness, and design.

#### To Validate

1. Make sure all PR Checks have passed (GitHub Actions, CircleCI, Code Climate, Snyk, etc).
2. Pull down all related branches.
3. Confirm all tests pass.

### CODE REVIEW

**Review by mdt2 — DISMISSED** [review]
This looks great! It is keyboard and VoiceOver accessible. One thing I've been looking into a solution for is when using VoiceOver, it says the label for the checkbox three times, so it says "Developer and 1 more item unchecked checkbox" and then when I go to the next item it says "Developer" and the next item says "Developer" again. While this still counts as accessible, it definitely would be annoying and inconvenient for a screen reader user, which is why I've been looking into a fix. I'll let you know if I find an improvement, but otherwise this looks good to me barring any rebasing that may need to be done because of #38

**Comment by theogainey** [comment]
> This looks great! It is keyboard and VoiceOver accessible. One thing I've been looking into a solution for is when using VoiceOver, it says the label for the checkbox three times, so it says "Developer and 1 more item unchecked checkbox" and then when I go to the next item it says "Developer" and the next item says "Developer" again. While this still counts as accessible, it definitely would be annoying and inconvenient for a screen reader user, which is why I've been looking into a fix. I'll let you know if I find an improvement, but otherwise this looks good to me barring any rebasing that may need to be done because of #38

I found a solution but its aria so I am not 100% on it. What I believe is happening is that the screen reader is reading the input, span, and pseudo element that gives us the cool slider affect. Adding these aria attributes seems to fix that but I have been conditioned to approach aria attributes with caution. So thoughts? 
```
 <label className="cmp-filter-toggle">
      <input
        className="cmp-filter-toggle__input"
        type={type}
        onChange={onChange}
        name={name}
        checked={checked}
        value={value}
        aria-label={label}
      />
      <span className="cmp-filter-toggle__text" aria-hidden="true" data-text={label}>{label}</span>
    </label>
```

**Comment by mdt2** [comment]
> > This looks great! It is keyboard and VoiceOver accessible. One thing I've been looking into a solution for is when using VoiceOver, it says the label for the checkbox three times, so it says "Developer and 1 more item unchecked checkbox" and then when I go to the next item it says "Developer" and the next item says "Developer" again. While this still counts as accessible, it definitely would be annoying and inconvenient for a screen reader user, which is why I've been looking into a fix. I'll let you know if I find an improvement, but otherwise this looks good to me barring any rebasing that may need to be done because of #38
> 
> I found a solution but its aria so I am not 100% on it. What I believe is happening is that the screen reader is reading the input, span, and pseudo element that gives us the cool slider affect. Adding these aria attributes seems to fix that but I have been conditioned to approach aria attributes with caution. So thoughts?
> 
> ```
>  <label className="cmp-filter-toggle">
>       <input
>         className="cmp-filter-toggle__input"
>         type={type}
>         onChange={onChange}
>         name={name}
>         checked={checked}
>         value={value}
>         aria-label={label}
>       />
>       <span className="cmp-filter-toggle__text" aria-hidden="true" data-text={label}>{label}</span>
>     </label>
> ```

This is great! It's working for me using VoiceOver 🎉 You're right to approach ARIA with some caution for two reasons I can think of:

1. Semantic HTML is always the best option if it can solve the problem
2. ARIA is not interpreted the same way by all assistive technologies (more on this in the spiel below)

So in this case, using ARIA attributes is ok and helpful from an accessibility standpoint because we're not really using HTML in the way it was intended by adding extra markup to make the checkboxes look how we want them to. 

Here's a little spiel that you may already know about from your accessibility lessons in the apprenticeship:
Ideally when we make an accessibility change like this, especially in the context of a client project, we would also test it at least using the NVDA screen reader (and _really_ ideally, we would have a person with disabilities test it for us, but that can be a rare opportunity so we just work to test it as best we can), but NVDA is only available on Windows, which I don't think either of us have access to. The reason for testing on multiple screen readers is that [they all work a little bit differently](https://www.oomphinc.com/inside-oomph/apples-and-oranges-all-screen-readers-are-not-created-equal/). Testing different screen readers is comparable to testing in multiple browsers. But given that we don't have access to Windows atm and that this is internal, I'm good with this solution!

File: styles/components/_filter-toggle.scss [hunk 1]
```diff
@@ -23,10 +23,21 @@
...
 
-    &:hover {
+    &::before {
+      display: block;
       background-color: color("white");
       color: color("black");
+      inset: 0;
+      position: absolute;
+      content: attr(data-text);
+      clip-path: polygon(0 0, 0 0, 0 100%, 0 100%);
```
Comment (arnest00) [inline styles/components/_filter-toggle.scss:35]: Nice!

**Review by mdt2 — APPROVED** [review]
dVdIu1HNxeKyqzkgPA

**Review by arnest00 — APPROVED** [review]

**Pull request merged**